src/ajax.js

```diff
@@ -54,8 +54,11 @@ export function ajax(url, callback, data, options = {}) {
     if (options.withCredentials) {
       x.withCredentials = true;
     } else {
-      x.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
-      x.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+      if (options.preflight !== false) {
+        x.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+      }
+      x.setRequestHeader('Content-Type',
+        options.contentType || 'application/json;charset=UTF-8');
     }
```

Comment:
this will also have to be updated, as content `other than application/x-www-form-urlencoded, multipart/form-data, or text/plain` requires a preflight request. In other words, `application/json` will trigger preflight.